source/s3_meta_request.c

```diff
@@ -868,6 +883,18 @@ static void s_s3_meta_request_send_request_finish(
     vtable->send_request_finish(vip_connection, stream, error_code);
 }
 
+void aws_s3_meta_request_finish_default(
+    struct aws_s3_meta_request *meta_request,
+    struct aws_s3_request *failed_request,
+    int error_code) {
+
+    if (failed_request != NULL) {
+        aws_s3_meta_request_finish(meta_request, failed_request, failed_request->send_data.response_status, error_code);
```

Comment:
Contents of the aws_s3_meta_request_finish should be able to become to the default implementation.  And then the status code is used for more than just the failed request, so we can't hide it from the interface.  (Let me know if that becomes an awkward issue!  We can probably make an improvement there, but I think it might be easier for this change if we just leave it as is.)